### DISCUSSION TITLE
Capture Routines Result in Syntax Error: Invalid or Unexpected Token

### DISCUSSION THREAD

**Discussion by FrankNJohnson** [chhoumann/quickadd#577] [Q&A]
I'm having a problem with the QuickAdd capture. I have three different capture routines set up which worked fine for awhile, but suddenly started returning an error message a few days ago. All of my QuickAdd templates continue to work fine.

For example, I have a capture that is designed to allow me to enter some text which will then be added in a particular place in my daily note with a timestemp.

The attached image shows how the capture is set up.

<img width="665" alt="quickadd-log-capture-error" src="https://github.com/chhoumann/quickadd/assets/116246915/6df56a4a-466d-4cb0-afe8-4009221fa45b">

A few days ago, running the capture started returning this error message:

"QuickAdd: (ERROR) Syntax Error: Invalid or unexpected token"

I did add the these plugins around the same time as the error started showing up: Pandoc Plugin, Advanced Tables, and Zotero. But disabling those plugins hasn't solved the issue.

In addition to disabling those plugins, I uninstalled and reinstalled QuickAdd, then copied over the data.json file from the previous installation to get my capture and template routines back.

Thanks in advance for any help any one might be able to offer - much appreciated!

**Comment by chhoumann** [comment]
Hi @FrankNJohnson, thanks for posting this. 

Do you by chance have Templater?

Also, would you be able to check if there are any QuickAdd-related error messages in the 'developer console'?
(cmd+ option + i on Mac and ctrl + shift + i for Windows).

**Comment by FrankNJohnson** [comment]
Thanks @chhoumann for the quick response!

I do have Templater installed.

Regarding the developer console, I suspect you don't need all of what I've put below, but trying to be complete ...

When I look at the developer console, I see a few QuickAdd errors - but they are all the same (matching the number of times I recently tried one of the capture routines):

```Quickadd: (ERROR) Syntax Error: Invalid or unexpected token
logError @ plugin:quickadd:17487```

Here are lines 17468-17502 in the sources panel:

```// src/logger/quickAddLogger.ts
var QuickAddLogger = class {
  formatOutputString(error) {
    return `QuickAdd: (${error.level}) ${error.message}`;
  }
  getQuickAddError(message, level) {
    return { message, level, time: Date.now() };
  }
};

// src/logger/consoleErrorLogger.ts
var ConsoleErrorLogger = class extends QuickAddLogger {
  constructor() {
    super(...arguments);
    this.ErrorLog = [];
  }
  logError(errorMsg) {
    const error = this.getQuickAddError(errorMsg, "ERROR" /* Error */);
    this.addMessageToErrorLog(error);
    console.error(this.formatOutputString(error));
  }
  logWarning(warningMsg) {
    const warning = this.getQuickAddError(warningMsg, "WARNING" /* Warning */);
    this.addMessageToErrorLog(warning);
    console.warn(this.formatOutputString(warning));
  }
  logMessage(logMsg) {
    const log2 = this.getQuickAddError(logMsg, "LOG" /* Log */);
    this.addMessageToErrorLog(log2);
    console.log(this.formatOutputString(log2));
  }
  addMessageToErrorLog(error) {
    this.ErrorLog.push(error);
  }
};```

Line 17487 is definitely the line with the error (little red circle with an "X" in it):

`console.error(this.formatOutputString(error)):`

Thanks again - I'm hoping there is a quick easy solution. My work processes are heavily dependent on QuickAdd! :)

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that. We never got to the bottom of it at the time, and thanks for the thorough console dig - the snippet you pasted was QuickAdd's logger printing the error rather than the code that caused it, which is why it didn't point anywhere useful.

This report is from a 2023 version, and the capture engine and format parsing it went through have been rewritten substantially since, so I can't reproduce or diagnose the original setup anymore. Closing as outdated rather than resolved.

If you (or anyone landing here) see "Syntax Error: Invalid or unexpected token" on a current QuickAdd, please open a fresh issue and include the capture's Format string and the full console stack trace - with those two things it's usually quick to pin down.
